### DISCUSSION TITLE
Download file from REST API

### DISCUSSION THREAD

**Discussion by NicolasBonet** [Budibase/budibase#8253] [Help]
Hello guys,

In our use case, we want to download a CSV with the data from a REST API service, I tried to set up a table and download using a button referencing the table, but that is not working (apparently it doesn't work with REST API Data: https://github.com/Budibase/budibase/issues/4895#issuecomment-1157525518). To proceed I created a middleware that only needs the JWT token in the headers; however, I tried two things and none of them worked:

- First I simply added a button that should execute the REST Call, this is correctly done, but Budibase ignores the header of the response and it does nothing with it.
- Second, I tried doing a navigateTo action to this GET request; however, that request is not done with the authenticated header.

So, I was wondering if any of you could please let me know how I could implement this.

Thank you!

**Comment by melohagan** [comment]
Hey @NicolasBonet 

Have you tried using the OAuthToken in the REST API headers? https://docs.budibase.com/docs/authentication-and-sso#using-the-oauth-token

**Reply by NicolasBonet** [reply]
Yes, I did, if I do that the request is successfully done, but the response is not interpreted by Budibase and no file is available to download.

**Reply by melohagan** [reply]
An alternative could possibly be to provide a link to the file rather than through a REST API?

**Reply by NicolasBonet** [reply]
Is that something we can do without S3? Do you've an example? It won't be ideal, but possibly a workaround for now!

**Comment by antoine-gu** [comment]
Similar here. I would like to get the data with this link into my bb
https://data.wien.gv.at/daten/geo?service=WFS&request=GetFeature&version=1.1.0&typeName=ogdwien:BAUMKATOGD&srsName=EPSG:4326&outputFormat=csv
